sqlmesh/dbt/model.py

```diff
@@ -181,16 +182,26 @@ def model_kind(self, context: DbtContext) -> ModelKind:
         """
         target = context.target
         materialization = self.model_materialization
+
+        # args common to all sqlmesh incremental kinds, regardless of materialization
+        incremental_kind_kwargs: t.Dict[str, t.Any] = {}
+        if self.on_schema_change:
+            incremental_kind_kwargs["on_destructive_change"] = (
+                OnDestructiveChange.ALLOW
+                if self.on_schema_change == "sync_all_columns"
+                else OnDestructiveChange.ERROR
```

Comment:
This doesn't look right to me. What informed this choice?

Reply:
Responded in slack thread https://tobikodata.slack.com/archives/C04DTU8RF4J/p1716262698103299